Added wrapper over some more parts of API

barnwal-paras opened the pull request.

barnwal-paras committed:
project upload

barnwal-paras committed:
added comment

barnwal-paras committed:
added xmloperations.cs

barnwal-paras committed:
added call and number search functionality

barnwal-paras committed:
correction in xmloperations

barnwal-paras committed:
Merge branch 'branch1' of https://github.com/barnwal-paras/restcomm-sdk-dotnet into branch1

barnwal-paras committed:
correction in xmloperation.cs

barnwal-paras committed:
added existing file

barnwal-paras committed:
Added CLient.cs

barnwal-paras committed:
Email added

barnwal-paras committed:
rearranged file

gvagenas commented:
@barnwal-paras lets work on one PR for now. Keep your work on one branch, unless if you have to experiment on something, and keep only one PR open. 

Please update the branches as needed, and close one of the two PRs.

Thanks
George

barnwal-paras committed:
unit tests and examples added

barnwal-paras committed:
separated unit testing and fixed issue with xmlresponse handling

barnwal-paras committed:
added  Examples

barnwal-paras committed:
Fixed some minor issue

gvagenas commented on src/AccountManager.cs:
```diff
@@ -0,0 +1,179 @@
+		public accountProperties Properties;
+
+
+		public static string baseurl="https://cloud.restcomm.com/restcomm/2012-04-24/";
```
@barnwal-paras base url should be configurable. The SDK will be used either with cloud.restcomm.com or any other Restcomm-Connect installation

gvagenas commented on src/AccountManager.cs:
```diff
@@ -0,0 +1,179 @@
+
+
+
+namespace RestComm
```
@barnwal-paras what is the convention for namespace? Can we use `org.restcomm.connect.sdk.dotnet` ?

gvagenas commented on src/AccountManager.cs:
```diff
@@ -0,0 +1,179 @@
+		{
+
+
+			RestClient client = new RestClient(baseurl+"Accounts/"+sid);
```
@barnwal-paras we should use the JSON endpoints all over the project

gvagenas reviewed (changes requested):
@barnwal-paras nice that you separated the unit tests from the actual implementation. 

following some comments for the next steps:

- Please add to gitignore the binaries and objects 
- Make Examples part of the solution, so if one open Restcomm.sln should have Examples also
- Change the solution name to Restcomm-Connect-DotNet-SDK or something similar that will be clear this is the dotnet sdk
- Make the solution a nuget so a developer can include it in his project
- Change the namespace to something that will be clear this is the dotnet sdk, such as `org.restcomm.connect.sdk.dotnet`
- Use the json endpoints
- Base URL must be passed as an argument to the main constructor of the SDK and NOT be hardcoded

The most important of all is the Unit Test should work with a mock of Restcomm-Connect and NOT using the cloud.restcomm.com. Ping me if you need help on that but its very important that you don't test against cloud.restcomm.com

I guess you are working on the following:
- Tutorial
- Documentation

barnwal-paras committed:
Added Gitignore and unit testing on mockserver

gitignore added and testing is done on mock server plus using json instead of xml endpoints.

barnwal-paras committed:
Removed Duplicate Files

barnwal-paras committed:
Json Deserialization

barnwal-paras committed:
added Unit-test for number search

gvagenas commented:
@barnwal-paras there must be a problem with the solution setup. When I open Restcomm.sln the classes in the src folder doesn't exist.
<img width="1440" alt="screen shot 2017-06-22 at 14 39 47" src="https://user-images.githubusercontent.com/588875/27432411-a788a618-5758-11e7-8b76-14ec317eb984.png">

Am I missing something? Can you please check?

Also please open issues for the missing items/tasks. Like this we merge this initial PR and your work from now on will be with PRs based on issues.

gvagenas commented:
Also provide README.md with the following:

- Give a short description of the project (1-2 lines)
- how to build the project
- how to run the tests
- dotnet version supported
- Reference any dependencies that might be needed to be added by a developer
- Provide instructions for where to download Restcomm and point to the restcomm documentation site
- Provide information how and where developers can open issues and feature requests

Also please provide example application and instructions for how to run it etc.

Imagine you have a dotnet developer but not familiar with Restcomm-Connect, that wants to use the library. Given that, we must provide any necessary information for him to get started.

barnwal-paras commented:
I am working on documentation and description . Will update that on github by tomorrow .
Can you confirm if you have latest version of the project ( https://github.com/barnwal-paras/restcomm-sdk-dotnet.git ) from branch1 .Project in master branch is not updated .

barnwal-paras committed:
Deleted old project file .

barnwal-paras commented:
My bad . Actually i forgot to delete duplicate project file  restcomm.csproj  .please open solution file (Restcomm-dot-net-sdk.sln)  or open Restcomm-dot-net-sdk.csproj inside Restcomm-dot-net-sdk folder to load the solution .

barnwal-paras committed:
Added project description in Read.md

gvagenas commented:
Ok great, checking now @barnwal-paras

gvagenas commented:
Great work @barnwal-paras 👍
